File: app/src/main/java/com/kent/university/privelt/ui/detailed/DetailedCardViewModel.java
/*
 *  This Source Code Form is subject to the terms of the Mozilla Public
 *  * License, v. 2.0. If a copy of the MPL was not distributed with this
 *  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package com.kent.university.privelt.ui.detailed;

import com.kent.university.privelt.model.UserData;
import com.kent.university.privelt.repositories.UserDataRepository;

import java.util.List;
import java.util.concurrent.Executor;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

public class DetailedCardViewModel extends ViewModel {
    private final UserDataRepository mUserDataRepository;
    private final Executor mExecutor;

    private LiveData<List<UserData>> userDatas;

    public DetailedCardViewModel(UserDataRepository mUserDataRepository, Executor mExecutor) {
        this.mUserDataRepository = mUserDataRepository;
        this.mExecutor = mExecutor;
    }

    void init(long serviceId) {
    }

    LiveData<List<UserData>> getUserDatasForService(long serviceId) {
        return userDatas;
    }
}

File: app/src/main/java/com/kent/university/privelt/ui/dashboard/card/detailed/DetailedCardViewModel.kt
/*
 *  This Source Code Form is subject to the terms of the Mozilla Public
 *  * License, v. 2.0. If a copy of the MPL was not distributed with this
 *  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.kent.university.privelt.ui.dashboard.card.detailed

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.kent.university.privelt.model.UserData
import com.kent.university.privelt.repositories.UserDataRepository
import java.util.concurrent.Executor

class DetailedCardViewModel(private val mUserDataRepository: UserDataRepository, private val mExecutor: Executor) : ViewModel() {
    private val userDatas: LiveData<List<UserData>>? = null
    fun init(serviceId: Long) {}
    fun getUserDatasForService(serviceId: Long): LiveData<List<UserData>>? {
        return userDatas
    }

}